uiowa/uiowa
# Admissions: Student Profile block revisited

# Updates

* Adapts parts of #2702 to solve for a more limited use case of providing filtering options in the student profile views block.
* Resolves https://github.com/uiowa/uiowa/issues/3138
* Resolves https://github.com/uiowa/uiowa/issues/3232

# How to test

- Pull and sync admissions
- Apply and Request Info Buttons should still be placed on AoS and working.
- Apply button should still be placed on student profiles and working.
- Provide featured and thumbnail images for a handful of student profiles.
- Block should still be placed on the student profile template with the same defaults as PROD. (9 items, "Student Profiles", no filtering)
- Block should be available to place by webmasters on page content type
- The filters (below), number of items and headline should be available in the block configuration.
  - Gender
  - Student Attributes
  - State
  - Country if Not US
- The block should not display if there are no valid profiles. Check this by filtering the content overview to just student profiles and unpublish all.
- Pick a student profile and publish it. Visit the full node. Block should still not display because the current profile should not be included.
- Pick another profile and publish it. The block should be visible on this and the previous node showing the other profile.
- Background image fallback for student profiles without an thumbnail.
- View should display random but is cached so you will need to run `cr` and reload the page to test or place different blocks with the same filtering
- Check other views allowed to be placed by webmaster because the views label is now hidden and replaced by the headline field.
